### DISCUSSION TITLE
How to convert data from PolyData to UniformGrid?

### DISCUSSION THREAD

**Discussion by lingchen1991** [pyvista/pyvista#4434] [Q&A] [answered]
I tried to get a 2D numpy array from a mesh by the function in https://github.com/pyvista/pyvista-support/issues/89. But it cannot work it seems my mesh is in PolyData while the function needs UniformGrid.

May I ask how to convert it?

**Comment by akaszynski** [comment, marked as answer]
Hello,

It's not possible to convert between a [PolyData](https://docs.pyvista.org/version/stable/api/core/_autosummary/pyvista.PolyData.html) object and a [UniformGrid](https://docs.pyvista.org/version/dev/api/core/_autosummary/pyvista.UniformGrid.html#pyvista-uniformgrid) as these objects organize their data differently. `PolyData` is composed of connected or disconnected faces, vertices, and/or lines, and a `UniformGrid` is just a set of cells with uniform spacing as defined by dimensions, origin, and spacing.

**Reply by banesullivan** [reply]
You could resample the polydata surface to a uniform grid -- this has lots of issues and the end result will be a rough approximation for most use cases
